Sources/AEPOptimize/Optimize.swift

```diff
@@ -226,12 +226,18 @@ public class Optimize: NSObject, Extension {
     /// The event is dispatched internally upon receiving an Edge content complete response for an update propositions request.
     /// - Parameter event: Optimize content complete event.
     private func processUpdatePropositionsCompleted(event: Event) {
+        let requestCompletedForEventId = event.data?[OptimizeConstants.EventDataKeys.COMPLETED_UPDATE_EVENT_ID] as? String ?? ""
+
         defer {
+            // remove completed event's ID from the request event IDs dictionary.
+            updateRequestEventIdsInProgress.removeValue(forKey: requestCompletedForEventId)
```

Comment:
if this method ends up reaching the `return` on line 248, `requestCompletedForEventId` will be an empty string and this call doesn't make sense. 

let's move this line back to where it was (out of the defer block) so it will only be called when we have an entry to remove.